pydra/engine/tests/test_task.py

```diff
@@ -986,6 +987,82 @@ def testfunc(a: int, b: float = 0.1) -> ty.NamedTuple("Output", [("out", float)]
     assert (tmpdir / funky.checksum / "messages.jsonld").exists()
 
 
+def test_audit_task(tmpdir):
+    @mark.task
+    def testfunc(a: int, b: float = 0.1) -> ty.NamedTuple("Output", [("out", float)]):
+        return a + b
+
+    from glob import glob
+
+    funky = testfunc(a=2, audit_flags=AuditFlag.PROV, messengers=FileMessenger())
+    funky.cache_dir = tmpdir
+    funky()
+    message_path = tmpdir / funky.checksum / "messages"
+    # go through each jsonld file in message_path and check if the label field exists
+    json_content = []
+    for file in glob(str(message_path) + "/*.jsonld"):
+        with open(file, "r") as f:
+            data = json.load(f)
+            if "label" in data:
+                json_content.append(True)
+                assert "Python Function" == data["label"]
+    assert any(json_content)
+
+
+def test_audit_shellcommandtask(tmpdir):
+    args = "-l"
+    shelly = ShellCommandTask(
+        name="shelly",
+        executable="ls",
+        args=args,
+        audit_flags=AuditFlag.PROV,
+        messengers=FileMessenger(),
+    )
+
+    from glob import glob
+
+    shelly.cache_dir = tmpdir
+    shelly()
+    message_path = tmpdir / shelly.checksum / "messages"
+    # go through each jsonld file in message_path and check if the label field exists
+    label_content = []
+    command_content = []
+
+    for file in glob(str(message_path) + "/*.jsonld"):
+        with open(file, "r") as f:
+            data = json.load(f)
+            if "label" in data:
+                label_content.append(True)
+            if "command" in data:
+                command_content.append(True)
+                assert "ls -l" == data["command"]
+
+    print(command_content)
+    assert any(label_content)
+
+
+def test_audit_task(tmpdir):
```

Comment:
you have two tests with the same name, so only the second is run, I believe this is the old version that should be removed